### ISSUE TITLE
Could not retrieve Message by index

### ISSUE THREAD

**Issue by rusmichal** [twilio/twilio-chat-demo-android#92]
Hey Twilio, we are using SDK in our implementation of chat feature. we are facing an issue related with media. The issue is:

`Caused by: java.lang.Throwable: Error 0:-4 Could not retrieve Message by index`
We are trying figure out what the error means but we can't find any advice in documentation. The error shows up when we are trying to get media from message. This is an example.:

```
channel.messages.getMessageByIndex(messageSid, object : CallbackListener<Message>() {
                            override fun onSuccess(message: Message) {
                                var outStream: ByteArrayOutputStream? = ByteArrayOutputStream(message.media.size.toInt())

                                message.media.download(outStream, object : StatusListener() {
                                    override fun onSuccess() {}

                                }, object : ProgressListener() {
                                    override fun onStarted() {}

                                    override fun onProgress(progress: Long) {}

                                    override fun onCompleted(messageSid: String?) {
                                        val inputStream = ByteArrayInputStream(outStream?.toByteArray())
                                        outStream = null
                                        it.onSuccess(inputStream)
                                    }
                                })
                            }

                            override fun onError(errorInfo: ErrorInfo?) {
                                it.onError(Throwable(errorInfo.toString()))
                            }
                        })
```
Once we call the method `channel.messages.getMessageByIndex(sid)` the error occurred. It just happens randomly. I can't find any pattern. I have no clue what it going with that and what can cause the issue. Do you have any idea? Does it mean the `sid` or media is not valid anymore?

#### Chat Android SDK

4.0.4

#### Android API

26

#### Android Device

Huawei P9

**Comment by berkus** [comment]
`.getMessageByIndex(messageSid`

Erm, sid is a string, index is a long. I'm confused, with which one are you calling?

Is there a message with such index in the channel? Is channel synchronized?

**Comment by rusmichal** [comment]
My bad. Yes we use `messageSid` it is a Long. This is `messageIndex` from `com.twilio.chat.Message`. I think the message exists. I can see channel id and index.

**Comment by rusmichal** [comment]
do you know what does this mean? `Error 0:-4`

**Comment by kontownik** [comment]
got the same issue, could you tell me more about `Error 0:-4`?  Looks like access is denied from Twilio side (sometimes reconnecting is fixing the issue).

**Comment by berkus** [comment]
`Error 0:-4` means `CannotGetMessageByIndex`.

This happens if returned message is null, which could happen if for example the channel messages have not synchronized properly or twilio request has timed out. Could you generate logs as described [here](https://github.com/twilio/twilio-chat-demo-android/blob/master/REPORT_BUGS.md) and post them?

**Comment by lpbaptista** [comment]
@berkus I've opened a support ticket (2012283) which so far I didn't receive any reply about a possible solution. Last reply was 17 of August which was saying that this would be escalated.

**Comment by berkus** [comment]
@lpbaptista this bug is fixed and will be released in the next SDK version.

**Comment by rusmonster** [comment]
As temporary workaround just call `getMessagesBefore(index, 1, ...)` instead of `getMessagesByIndex(index, ...`

**Comment by lpbaptista** [comment]
@rusmonster we don't actually use the method `getMessageByIndex` we only retrieve messages from `channel.messages?.getMessagesBefore(index, limit, object : CallbackListener<List<Message>>()...`

**Comment by astamato** [comment]
Tested the workaround suggested above and works like a charm.

Although, just a comment. From the latest docs for the Android SDK:

* http://media.twiliocdn.com/sdk/android/chat/releases/4.2.3/docs/classcom_1_1twilio_1_1chat_1_1Messages.html#a4323575af08a2bb1cbf3195a0709d105

Never does it say that includes message with the given index (says before given index explicitly). 

Alternatively:

* https://www.twilio.com/docs/chat/result-paging#messages-android-ios (specifies including anchor message)
* https://media.twiliocdn.com/sdk/ios/chat/releases/2.6.2/docs/Classes/TCHMessages.html#//api/name/getMessagesBefore:withCount:completion: (specifies including and prior)

Maybe Android docs can be updated? Cheers.

**Comment by rusmonster** [comment]
fixed in sdk 4.2.4
